src/main/java/duke/chatbot/util/Parser.java

```diff
@@ -0,0 +1,159 @@
+package duke.chatbot.util;
+
+import static duke.chatbot.common.DateFormat.DATE_TIME_INPUT_FORMAT;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.chatbot.command.AddDeadlineCommand;
+import duke.chatbot.command.AddEventCommand;
+import duke.chatbot.command.AddToDoCommand;
+import duke.chatbot.command.CheckDateCommand;
+import duke.chatbot.command.Command;
+import duke.chatbot.command.DeleteCommand;
+import duke.chatbot.command.ExitCommand;
+import duke.chatbot.command.FindCommand;
+import duke.chatbot.command.ListCommand;
+import duke.chatbot.command.MarkCommand;
+import duke.chatbot.command.UnmarkCommand;
+import duke.chatbot.data.exception.InvalidInputException;
+import duke.chatbot.data.exception.InvalidStorageFileException;
+import duke.chatbot.data.task.Deadline;
+import duke.chatbot.data.task.Event;
+import duke.chatbot.data.task.TaskList;
+import duke.chatbot.data.task.ToDo;
+
```

Comment:
really love the use of OOP to abstract everything but maybe these 30 lines of import statements might indicate over-abstraction? not too big of an issue either way